worf/serializers.py

```diff
@@ -1,35 +1,64 @@
 import json
+import marshmallow
 
 from django.core.exceptions import ImproperlyConfigured
 
+from worf import fields  # noqa: F401
+from worf.casing import snake_to_camel
+
 
 def deserialize(response):
     return json.loads(response.content.decode("UTF-8"))
 
 
-class Serializer:
-    def create(self):
-        return []
+class Serializer(marshmallow.Schema):
+    """
+    Serializer base class with sensible defaults for typically read-only fields.
+
+    Worf serializers are basically marshmallow schemas with better support for
+    Django models, and some extra opinions around field naming and ordering.
+
+    The read/write methods are relics of our interim serialization strategy
+    they're analogous with marshmallow dump/load, but a bit less flexible.
+
+    Marshmallow is primarily used for field filtering right now, worf coerces
+    values itself, but we might delegated more to marshmallow later.
+    """
+
+    id = fields.Function(lambda obj: obj.pk, dump_only=True)
+    created_at = fields.DateTime(dump_only=True)
+    deleted_at = fields.DateTime(dump_only=True)
+    updated_at = fields.DateTime(dump_only=True)
+
```

Comment:
I think your assumptions around created_at deleted_at etc are valid and make sense. The note around id, created_at, deleted_at, and updated_at being read-only by default ought to find its way into the docs because someone will lose their mind trying to figure out why it doesn't work to write. I'm not sure if Worf should set these fields to read-only or leave that to the users to implement, but I'm fine starting here. 

I imagine all it takes to override this is to set it on the serializer that inherits, so no worries

Reply:
I decided not to set these by default but instead add a mechanism for settings these via django `settings`, along with any other default meta you might want to apply, saving clients from creating an abstract base serializer, or needing to extend metaclasses to do it, which also means I can leave the implicit `id` alone and let it coerce to String or Int.